src/dev/flang/ast/Impl.java

```diff
@@ -522,13 +525,42 @@ AbstractType inferredType(Resolution res, AbstractFeature f)
     return switch (_kind)
       {
       case FieldDef    -> _initialValue.typeIfKnown();
-      case RoutineDef  -> _code.typeIfKnown();
+      case RoutineDef  -> typeFromRedefinedFeatures(f);
       case FieldActual -> typeFromInitialValues(res, f, false);
       default -> throw new Error("missing case "+_kind);
       };
   }
 
 
+  AbstractType typeFromRedefinedFeatures(AbstractFeature f)
```

Comment:
This should be named `typeFromRedefinedOrCode` or similar, in most cases `f.redefines()` will be empty and the type will be taken from the code.

Please add a comment to this. You may need to add `res` as an argument.